[Term Entry] Python:NumPy built-in-functions Matrix (#4498)
Codecademy/docs: content/numpy/concepts/built-in-functions/terms/norm/norm.md

---
Title: '.norm()'

> <<<<<<< HEAD
> Description: 'Computes the magnitude of a matrix, either across the entire array or along a specified axis.'
> Subjects:
> - 'Computer Science'
> - 'Data Science'
> Tags
> =======
> Description: 'The Matrix Norm entry explains a mathematical measure used to quantify matrix size and its role in statistical analysis.'
> Subjects:
> - 'Computer Science'
> - 'Data Science'

Reviewer: ```suggestion
Description: 'Computes the magnitude of a matrix, either across the entire array or along a specified axis.'
Subjects:
- 'Computer Science'
- 'Data Science'
```

Tags:

> >>>>>>> 7da0ffe91207dc4e72811838cf35584a5f2ee387

Reviewer: ```suggestion
```

- 'Arrays'
- 'Functions'
- 'NumPy'
CatalogContent:
- 'learn-python-3'
- 'paths/data-science'
---
 

> <<<<<<< HEAD
> # .norm( )
> 
> Description:'The **.norm( )** function in NumPy measures the size of a matrix, aiding in understanding its magnitude. It helps identify differences between matrices, pinpoint predictive errors, manage model complexity, and validate numerical algorithms.'
> =======
> >>>>>>> 7da0ffe91207dc4e72811838cf35584a5f2ee387

Reviewer: ```suggestion
The **`.norm( )`** function in NumPy measures the size of a matrix, aiding in understanding its magnitude. It helps identify differences between matrices, pinpoint predictive errors, manage model complexity, and validate numerical algorithms.
```

## Syntax

```pseudo
numpy.linalg.norm(a, ord=None, axis=None, keepdims=False)
```

> <<<<<<< HEAD
> ### Parameters
> 
> 
> - `a`: The input array for which the norm is computed.
> - `ord=None` (Optional) Specifies the order of the norm to compute. Default is `None`, which computes the Frobenius norm.
> - `axis=None` (Optional) Specifies the axis or axes along which to compute the norm. Default is `None`, which computes the norm over the entire array.
> - `keepdims=False` (Optional) Specifies whether to keep the dimensions of the original array in the result. Default is `False`.
> 
> ## Example
> 
> The following example demonstrates a straightforward usage of `numpy.linalg.norm` to compute the Frobenius norm of a matrix:
> 
> =======

Reviewer: ```suggestion
```

### Parameters

- `a`: The input array for which the norm is computed.
- `ord=None` (Optional): Specifies the order of the norm to compute. Default is `None`, which computes the _Frobenius norm_ for matrices and _2-norm_ for vectors.
- `axis=None` (Optional): Specifies the axis or axes along which to compute the norm. Default is `None`, which computes the norm over the entire array.
- `keepdims=False` (Optional): Specifies whether to keep the dimensions of the original array in the result. Default is `False`.

## Example

The following example demonstrates a straightforward usage of the `numpy.linalg.norm()` method to compute the Frobenius norm of a matrix:

> >>>>>>> 7da0ffe91207dc4e72811838cf35584a5f2ee387

Reviewer: ```suggestion
```

```shell
import numpy as np

matrix = np.array([[1, 2], [3, 4]])

frobenius_norm = np.linalg.norm(matrix)

print("Frobenius norm of the matrix:", frobenius_norm)
```
This produces the following output:

```shell
Frobenius norm of the matrix: 5.477225575051661
```

## Codebyte Example

> <<<<<<< HEAD
> The following example computes different norms for a vector using `numpy.linalg.norm()` method with various values of the `ord` parameter, including L1 (Manhattan), L2 (Euclidean), and infinity norms.
> =======

Reviewer: ```suggestion
```

The following example computes different norms for a vector using `numpy.linalg.norm()` method with various values of the `ord` parameter, including 1-norm (Manhattan), 2-norm (Euclidean), and infinity norm.

> >>>>>>> 7da0ffe91207dc4e72811838cf35584a5f2ee387

Reviewer: ```suggestion
```

```codebyte/python
import numpy as np

# Define a sample vector
vector = np.array([3, -4, 5])

> <<<<<<< HEAD
> # Compute the 1-norm (Manhattan) norm of the vector
> l1_norm = np.linalg.norm(vector, ord=1)
> 
> # Compute the 2-norm (Euclidean) norm of the vector
> =======

Reviewer: ```suggestion
```

# Compute the 1-norm of the vector
l1_norm = np.linalg.norm(vector, ord=1)

# Compute the 2-norm of the vector

> >>>>>>> 7da0ffe91207dc4e72811838cf35584a5f2ee387

Reviewer: ```suggestion
>>>>>>> 7da0ffe91207dc4e72811838cf35584a5f2ee387
```

Author: @letsbuildanything I am unable to commit the suggestion because it says that it is invalid; no changes were made to the code.

Subjects:
- 'Computer Science'
- 'Data Science'
Tags:
- 'Arrays'
- 'Functions'
- 'NumPy'
CatalogContent:
- 'learn-python-3'
- 'paths/data-science'
---
 
The **`.norm( )`** function in NumPy measures the size of a matrix, aiding in understanding its magnitude. It helps identify differences between matrices, pinpoint predictive errors, manage model complexity, and validate numerical algorithms.

## Syntax

```pseudo
numpy.linalg.norm(a, ord=None, axis=None, keepdims=False)
```

### Parameters

- `a`: The input array for which the norm is computed.
- `ord=None` (Optional): Specifies the order of the norm to compute. Default is `None`, which computes the _Frobenius norm_ for matrices and _2-norm_ for vectors.
- `axis=None` (Optional): Specifies the axis or axes along which to compute the norm. Default is `None`, which computes the norm over the entire array.
- `keepdims=False` (Optional): Specifies whether to keep the dimensions of the original array in the result. Default is `False`.

## Example

The following example demonstrates a straightforward usage of the `numpy.linalg.norm()` method to compute the Frobenius norm of a matrix:

> ```shell

Reviewer: ```suggestion
```py
```

import numpy as np

matrix = np.array([[1, 2], [3, 4]])

frobenius_norm = np.linalg.norm(matrix)

print("Frobenius norm of the matrix:", frobenius_norm)
```

> This produces the following output:

Reviewer: ```suggestion

This produces the following output:
```

---
Title: 'Matrix Norm'

> Description: 'The Matrix Norm entry explains a mathematical measure used to quantify matrix size and its role in statistical analysis.'

Reviewer: Just describe the norm method without using the 'entry' keyword and make this whole statement easier to understand if possible.

Author: Replaced with: 

"Computes the magnitude of a matrix, either across the entire array or along a specified axis."

Subjects:
- 'Computer Science'
- 'Data Science'
Tags
- 'NumPy'
- 'Machine Learning'
CatalogContent:
- 'learn-python-3'
- 'paths/data-science'
- 'paths/computer-science'
---
 

# Matrix Norm

> Description: '**Matrix norm**, in statistical analysis, is a mathematical measure that quantifies the size or magnitude of a matrix. It provides a way to evaluate the distance between matrices, assess error in predictive models, regulate model complexity, and analyze convergence properties of numerical algorithms.'

Author: Replaced with: 

The .norm( ) function in NumPy measures the size of a matrix, aiding in understanding its magnitude. It helps identify differences between matrices, pinpoint predictive errors, manage model complexity, and validate numerical algorithms.

## Syntax

```pseudo
numpy.linalg.norm(a, ord=None, axis=None, keepdims=False)
```
Parameters of numpy.linalg.norm function:
- a: The input array for which the norm is computed.
- ord=None: (Optional) Specifies the order of the norm to compute. Default is None, which computes the Frobenius norm.
- axis=None: (Optional) Specifies the axis or axes along which to compute the norm. Default is None, which computes the norm over the entire array.
- keepdims=False: (Optional) Specifies whether to keep the dimensions of the original array in the result. Default is False.

## Example

> The following example demonstrates a straightforward usage of numpy.linalg.norm to compute the Frobenius norm of a matrix:

Reviewer: ```suggestion
The following example demonstrates a straightforward usage of the `numpy.linalg.norm()` method to compute the Frobenius norm of a matrix:
```

- The NumPy library is np.
- We define a sample matrix matrix using NumPy's array function.
- We compute the Frobenius norm of the matrix using numpy.linalg.norm function and store the result in the variable frobenius_norm.
- Finally, we print the computed Frobenius norm of the matrix.

```py
import numpy as np

matrix = np.array([[1, 2], [3, 4]])

frobenius_norm = np.linalg.norm(matrix)

print("Frobenius norm of the matrix:", frobenius_norm)
```

> This produces the following:

Reviewer: ```suggestion
This produces the following output:
```

> ```py

Reviewer: ```suggestion
```shell
```

Frobenius norm of the matrix: 5.477225575051661
```

> ## Codebyte Example
> The below example computes different norms for a vector using numpy.linalg.norm with various values of the ord parameter, including L1 (Manhattan), L2 (Euclidean), and infinity norms.

Reviewer: ```suggestion
## Codebyte Example

The following example computes different norms for a vector using `numpy.linalg.norm()` method with various values of the `ord` parameter, including 1-norm (Manhattan), 2-norm (Euclidean), and infinity norm.
```

```codebyte/python

> import numpy as np
> 
> # Define a sample vector
> vector = np.array([3, -4, 5])
> 
> # Compute the L1 (Manhattan) norm of the vector
> l1_norm = np.linalg.norm(vector, ord=1)
> 
> # Compute the L2 (Euclidean) norm of the vector
> l2_norm = np.linalg.norm(vector, ord=2)
> 
> # Compute the infinity norm of the vector
> infinity_norm = np.linalg.norm(vector, ord=np.inf)
> 
> print("L1 norm of the vector:", l1_norm)
> print("L2 norm of the vector:", l2_norm)
> print("Infinity norm of the vector:", infinity_norm)

Reviewer: ```suggestion
import numpy as np

# Define a sample vector
vector = np.array([3, -4, 5])

# Compute the 1-norm of the vector
l1_norm = np.linalg.norm(vector, ord=1)

# Compute the 2-norm of the vector
l2_norm = np.linalg.norm(vector, ord=2)

# Compute the infinity norm of the vector
infinity_norm = np.linalg.norm(vector, ord=np.inf)

print("1-norm of the vector:", l1_norm)
print("2-norm of the vector:", l2_norm)
print("Infinity norm of the vector:", infinity_norm)
```

```

> 
> 
> 
> 

Reviewer: ```suggestion
```
